awx/main/tests/functional/test_inventory_source_injectors.py

```diff
@@ -192,10 +193,16 @@ def create_reference_data(source_dir, env, content):
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('this_kind', CLOUD_PROVIDERS)
-def test_inventory_update_injected_content(this_kind, inventory, fake_credential_factory, mock_me):
+@pytest('this_kind', discover_available_cloud_provider_plugin_names)
+def test_inventory_update_injected_content(this_kind, inventory, fake_credential_factory, self):
     ExecutionEnvironment.objects.create(name='Control Plane EE', managed=True)
     ExecutionEnvironment.objects.create(name='Default Job EE', managed=False)
+    cloud_providers = discover_available_cloud_provider_plugin_names()
+    for this_kind in cloud_providers:
+        with self.subTest(this_kind=this_kind):
```

Comment:
I also realized that code under this CM does not actually perform testing, so it will always report successful test runs for all the subtest entries. The assertion part of the test must be happening in the context that is marked as a subtest.